fix: null pointer exception error fields [IDE-1581]

acke opened the pull request:
### Description

In `getNewIacTreeNodeText`, when `currentIacError` is not null, the code tries to access `currentSnykCodeError` (line 545), which may be null, causing a NullPointerException. Similarly, in `getNewOssTreeNodeText` (line 594), when `realError` is true, it accesses `currentSnykCodeError` instead of `currentOssError`. Both should use the appropriate error field for their respective product types. @code-client-go/src/main/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanel.kt:544-545 @code-client-go/src/main/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanel.kt:593-594 

### Checklist

- [ ] Read and understood the [Code of Conduct](https://github.com/snyk/snyk-intellij-plugin/blob/master/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/snyk/snyk-intellij-plugin/blob/master/CONTRIBUTING.md).
- [ ] Tests added and all succeed
- [ ] Linted
- [ ] CHANGELOG.md updated
- [ ] README.md updated, if user-facing

### Screenshots / GIFs

_Visuals that may help the reviewer. Please add screenshots for any UI change. GIFs are most welcome!_

acke committed:
fix: use correct error fields in tree node text methods

Fix NullPointerException issues in getNewIacTreeNodeText and getNewOssTreeNodeText:
- In getNewIacTreeNodeText: use currentIacError instead of currentSnykCodeError when currentIacError is not null
- In getNewOssTreeNodeText: use currentOssError instead of currentSnykCodeError when realError is true

Both methods were accessing the wrong error field, which could be null and cause NullPointerExceptions.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/snyk/snyk-intellij-plugin?pullRequest=745) <br/>All committers have signed the CLA.

rrama reviewed (approved):
LGTM!
I left a few things I was wondering about which are unrelated to your fix, but if you know the answers I would be interested to know, if not then ehh.

acke committed:
Merge branch 'master' into fix/null-pointer-exception-error-fields

acke committed:
fix: improve error handling consistency in tree node text

- Fixed OSS to check errors before scanning status (matching IAC and Code Security)
- Fixed IAC to filter errors when no supported files found (matching OSS behavior)
- Updated CHANGELOG with 2.18.1 fixes
